File: clients/go/model/player_view.go
package model

import (
	"io"

	mStream "../stream"
)

type PlayerView struct {
	MyId int32
	Game *Game
}

func NewPlayerView(myId int32, game *Game) *PlayerView {
	return &PlayerView{
		MyId: myId,
		Game: game,
	}
}
func ReadPlayerView(reader io.Reader) *PlayerView {
	return &PlayerView{
		MyId: mStream.ReadInt32(reader),
		Game: ReadGame(reader),
	}
}
func (value *PlayerView) Write(writer io.Writer) {
	mStream.WriteInt32(writer, value.MyId)
	value.Game.Write(writer)
}

File: clients/cpp/model/PlayerView.cpp
#include "PlayerView.hpp"

PlayerView::PlayerView() { }
PlayerView::PlayerView(int myId, Game game) : myId(myId), game(game) { }
PlayerView PlayerView::readFrom(InputStream& stream) {
    PlayerView result;
    result.myId = stream.readInt();
    result.game = Game::readFrom(stream);
    return result;
}
void PlayerView::writeTo(OutputStream& stream) const {
    stream.write(myId);
    game.writeTo(stream);
}
std::string PlayerView::toString() const {
    return std::string("PlayerView") + "(" +
        std::to_string(myId) +
        game.toString() +
        ")";
}
